pkg/networkservice/common/excludedprefixes/server_test.go

```diff
@@ -84,13 +85,40 @@ func TestCheckReloadedPrefixes(t *testing.T) {
 	}, time.Second*15, time.Millisecond*100)
 }
 
+func TestWaitForFile(t *testing.T) {
+	prefixes := []string{"172.16.1.0/24", "10.96.0.0/12"}
+
+	testConfig := strings.Join(append([]string{"prefixes:"}, prefixes...), "\n- ")
+	configPath := filepath.Join(os.TempDir(), "excluded_prefixes.yaml")
+	defer func() { _ = os.Remove(configPath) }()
+
+	chain := next.NewNetworkServiceServer(excludedprefixes.NewServer(context.Background(),
+		excludedprefixes.WithConfigPath(configPath)))
+	req := &networkservice.NetworkServiceRequest{
+		Connection: &networkservice.Connection{
+			Context: &networkservice.ConnectionContext{},
+		},
+	}
+
+	time.AfterFunc(time.Millisecond*300, func() {
+		writingToFile(t, testConfig, configPath)
+		t.Logf("File was created")
+	})
+
+	require.Eventually(t, func() bool {
+		_, err := chain.Request(context.Background(), req)
+		require.NoError(t, err)
+		return reflect.DeepEqual(req.GetConnection().GetContext().GetIpContext().GetExcludedPrefixes(), prefixes)
+	}, time.Second, time.Millisecond*100)
```

Comment:
Test was unstable, because we don't handle case, when file was created after file read in ```init()``` and before fileWatcher creation in ```utils.watchFile()```. So I added `onChanged()` before select in `utils.watchFile()` to ensure, that we won't miss Create event.